### PR TITLE
[4.0] Add default stages/transitions when creating a new workflow

### PR DESCRIPTION
**Pull request by bembelimen**
Pull Request for Issue #21819 .

### Summary of Changes
When creating a new workflow now all default stages (publish, unpublished, trashed, archived) and transitions (unpublish, publish, trash, archive) will be created (instead of only "published").

(And yes, transitions are not translated yet, will come after #21939)

### Testing Instructions
- Go to Content => Workflows
- Create a new workflow
- Check the stages created with the workflow
- Check the transitions created with the workflow

### Expected result
- 4 stages (unpublished, published, trashed, archived)
- 4 transitions (unpublish, publish, trash, archive)

### Actual result
- only one new stage (published)

### CODE REVIEW

File: administrator/components/com_workflow/Model/WorkflowModel.php [hunk 1]
```diff
@@ -104,21 +105,64 @@ public function save($data)
 
 		$result = parent::save($data);
 
-		// Create a default stage
+		// Create a default stages/transitions
```
Comment (brianteeman) [inline administrator/components/com_workflow/Model/WorkflowModel.php]: very minor but ==>
 Create default stages/transitions

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on c2ad93970ebd366cd4ab0f51a2cf0c77d02cd32c<br /><br />perfect thanks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21963">issues.joomla.org/tracker/joomla-cms/21963</a>.</sub>

**Review by brianteeman — DISMISSED** [review]

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on 2b68b26de9a2c490898b6d347050531d69de7ef1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21963">issues.joomla.org/tracker/joomla-cms/21963</a>.</sub>

**Comment by ghost** [comment]
@brianteeman can you please retest?

**Comment by zero-24** [comment]
Moving to RTC as the only change after Brians test was the inline comment changed. Thanks!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21963">issues.joomla.org/tracker/joomla-cms/21963</a>.</sub>

**Comment by wilsonge** [comment]
I'm not convinced by default we should have an archived state in the default flow. but merging for now

**Pull request merged**